Ncsdk 3653 shell bt nus update (#2851)
nrfconnect/sdk-nrf: samples/bluetooth/shell_bt_nus/README.rst

Requirements
************

The sample supports the following development kits:

.. include:: /includes/boardname_tables/sample_boardnames.txt
   :start-after: set1_start
   :end-before: set1_end

You also need an additional nRF52 DK (PCA10040) for connecting with ``bt_nus_shell.py``.
Alternatively, you can use :ref:`ble_console_readme` for connecting, using Linux only.

Building and running
********************

.. |sample path| replace:: :file:`samples/bluetooth/shell_bt_nus`

.. include:: /includes/build_and_run.txt

Testing using shell_bt_nus
==========================

.. include:: ../../../include/shell/shell_bt_nus.rst
   :start-after: testing_bt_nus_shell_intro_start
   :end-before: testing_bt_nus_shell_intro_end

After programming the sample to your development kits, test it by performing the following steps:

1. Start a terminal emulator like PuTTY and connect through UART to the ``shell_bt_nus`` application running on the development kit to check the log.
   See :ref:`gs_testing` for more information on how to connect with PuTTY through UART.
#. On the host PC, install *Python 2*.

> #. Run the following command to install the |BLE| driver: ``pip2 install pc_ble_driver_py``.

Reviewer: ```suggestion
#. Run the following command to install the |BLE| driver:

   .. code-block:: console
   
      pip2 install pc_ble_driver_py
       
```

Author: fixed

> #. Run the following command to install :ref:`nRF Util`: ``pip2 install --user nrfutil==4.0.0``.

Reviewer: ```suggestion
#. Run the following command to install `nRF Util`_:

   .. code-block:: console
   
      pip2 install --user nrfutil==4.0.0
    
```

Author: fixed

#. Connect the nRF52 board meant for connecting with ``bt_nus_shell.py`` to your PC.
#. Start the ``bt_nus_shell.py`` script with the correct parameters, for example::

> 
>        bt_nus_shell.py --name BT_NUS_shell --com COM237 --family NRF52 --snr 682560213

Reviewer: ```suggestion

   .. code-block: console

      bt_nus_shell.py --name BT_NUS_shell --com COM237 --family NRF52 --snr 682560213

```

Author: fixed

> #. Open a terminal emulator like PuTTY and set the Connection Type to ``Raw`` and the  Destination Address to ``127.0.0.1:8889``.

Reviewer: ```suggestion
#. Open a terminal emulator like PuTTY and set the Connection Type to :guilabel:`Raw` and the  Destination Address to ``127.0.0.1:8889``.
```

Author: fixed

> #. Press Enter in the terminal.

Reviewer: Is this Enter or Open?

Author: I think it is actually both, press guilabel-open to actually open the terminal, then press enter in the terminal.
@nordic-krch Could you check this one?

Reviewer 2: it's pressing `enter`

Author: ok, so I rephrased it

> * One of the following development boards:

Reviewer: ```suggestion
The sample supports the following development kits:
```
This is for consistency with other samples, as implemented by Uma.

Author: fixed

.. include:: /includes/boardname_tables/sample_boardnames.txt
   :start-after: set1_start
   :end-before: set1_end

> * A second nRF52 Development Kit board (PCA10040) for connecting with bt_nus_shell.py.

Reviewer: ```suggestion
You also need an additional nRF52 DK (PCA10040) for connecting with ``bt_nus_shell.py``.
```

Author: fixed

>   Alternatively, you can use :ref:`ble_console_readme` for connecting, using Linux only.

Reviewer: Does this "alternatively" mean I don't need the nRF52 board anymore?

Reviewer 2: yes

Author: fixed

Building and running
********************

> This sample can be found under :file:`samples/bluetooth/shell_bt_nus` in the |NCS| folder structure.

Reviewer: ```suggestion
.. |sample path| replace:: :file:`samples/zigbee/light_bulb`
```

Author: Hey @greg-fer, are you sure about this suggestion? The original path to the sample seems to be correct.

Reviewer: Sorry, forgot to update the path in the suggestion. See the source of other samples -- this formula is standard for most of them.

Author: fixed

> See :ref:`gs_programming` for information about how to build and program the application.

Reviewer: ```suggestion
.. include:: /includes/build_and_run.txt
```

Author: fixed

Testing
=======

> After programming the sample to your board, test it by connecting to it and sending some shell commands.

Reviewer: With the recent edits and intro in line 40, this line looks like not needed anymore.

Author: fixed

> Perform the following steps to use the ``bt_nus_shell.py`` script:

Reviewer: ```suggestion
After programming the sample to your development kits, test it by performing the following steps:
```
This is the standard intro line to all the samples. Building and programming is already covered in line 33.

Author: fixed

> 1. Compile the application ``shell_bt_nus`` and flash it to one of the Nordic Development Kit listed in the requirements section.

Reviewer: This line is duplicating info from line 33.

Author: fixed

> #. Start a terminal or PuTTY and connect through UART to the ``shell_bt_nus`` application to check the log.

Reviewer: Shouldn't this line link to https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/gs_testing.html#putty , like many other samples?

Reviewer 2: yes, it's the same as in other apps so probably yes

Author: fixed

> #. On the host PC, install python2.

Reviewer: Is python2 really needed? If so, this needs to be mentioned in Requirements.

Reviewer 2: yes, it's needed

> #. Run ``pip2 install pc_ble_driver_py`` and ``pip2 install --user nrfutil==4.0.0``

Reviewer: Split into two steps maybe? With some explanation:
```
Run the following commands to install the |BLE| driver:
```
and so on...

Author: fixed

> #. Connect the 2nd nRF52 Development Kit board (PCA10040) to your PC.

Reviewer: This is only valid if the user uses an nRF52 DK for the first board. You can work around this by giving this board a descriptive name in Requirements, for example:
```suggestion
#. Connect the nRF52 board meant for connecting with ``bt_nus_shell.py`` to your PC.
```

Author: fixed

#. Start the ``bt_nus_shell.py`` script with the correct parameters, for example::

       bt_nus_shell.py --name BT_NUS_shell --com COM237 --family NRF52 --snr 682560213

> #. Open a terminal or PuTTY and set the Connection Type to ``Raw`` and the  Destination Address to ``127.0.0.1:8889``.

Reviewer: Can this part be still propagated from the other file? Otherwise we're going to have diverging procedures and can miss updates to either file.

Reviewer 2: fyi, this setup is unique to this sample.

Author: @nordic-krch Isn't this sample the same one mentioned here? https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/include/shell/shell_bt_nus.html#shell-bt-nus-readme

> #. Observe that the ``BSP_INDICATE_CONNECTED`` state is indicated.

Reviewer: Where is this indicated? If it an LED blinking or being solid, this needs to be described in User interface section.

Reviewer 2: Checked now that leds are not used to indicate the state. Only log message `Connected` is printed on the console.

>    Observe that both the ``shell_bt_nus`` and the ``bt_nus_shell.py`` connect.

Reviewer: How can I observe this? What is the visual hint or a console hint?

Reviewer 2: actually, this step can be removed. Once board indicates active connection (log message) then after pressing enter in the terminal prompt will be printed.

Author: fixed

#. Press Enter in the terminal.
   A console prompt is displayed.

> #. Enter the commands that you want to execute, like ``help``, ``echo``, and ``stats``, on the remote shell and check the log.

Reviewer: Can you provide an example of one of the commands? Where can I find the log?

> #. Enter the commands that you want to execute, like ``help``, ``echo``, and ``stats``, on the remote shell and check the log.

Reviewer: Overall, is it OK that after expanding the testing section, the overview and the intro of the sample stay exactly the same as before?

Author: I expanded it mentioning the additional info listed here about what it appears to be the same sample:
https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/include/shell/shell_bt_nus.html#shell-bt-nus-readme

Waiting for confirmation from @nordic-krch if this is actually correct. The steps could also be the same (and they are written in 2 different ways, so we could choose the one that looks best.

Author: @nordic-krch Could you please check this one?

Author: A lot of edits since this comment. I'm marking it resolved as the text changed quite a bit.

> Alternatively, use the :ref:`ble_console_readme` for testing.

Reviewer: How can I do this? Are the information about how to do this on the page linked?

Author: fixed
